common/azure-common-test/src/main/java/com/azure/common/test/utils/SdkContext.java

```diff
@@ -3,58 +3,151 @@
 
 package com.azure.common.test.utils;
 
-import com.azure.common.test.TestMode;
-import com.azure.common.test.models.RecordedData;
+import com.azure.common.http.HttpHeader;
+import com.azure.common.http.HttpResponse;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
 
-import java.util.Objects;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+import static com.azure.common.test.utils.TestConstant.DEFAULT_BUFFER_LENGTH;
 
 /**
  * The class to contain the common test methods for testing SDK.
  */
-public class SdkContext {
-    private final ResourceNamer nameGenerator;
-
-    /**
-     * Creates an SDK context that keeps track of variable names for the recorded data.
-     *
-     * <ul>
-     *     <li>If the {@code testMode} is {@link TestMode#PLAYBACK}, this will generate random variables by reading from
-     *     {@code recordedData}.</li>
-     *     <li>If the {@code testMode} is {@link TestMode#RECORD}, this will generate random variables and write them to
-     *     {@code recordedData}.</li>
-     * </ul>
-     *
-     * @param testMode The test for this context.
-     * @param recordedData The data to persist or read any variables names to or from.
-     */
-    public SdkContext(TestMode testMode, RecordedData recordedData) {
-        Objects.requireNonNull(recordedData);
-        nameGenerator =  new TestResourceNamer("", testMode, recordedData);
-    }
+public final class SdkContext {
 
     /**
      * Gets a random name.
      *
+     * @param namer the test resource random namer which is related to recorded data
      * @param prefix the prefix to be used if possible
      * @param maxLen the maximum length for the random generated name
      * @return the random name
      */
-    public String randomResourceName(String prefix, int maxLen) {
-        return nameGenerator.randomName(prefix, maxLen);
+    public static String randomTestResourceName(TestResourceNamer namer, String prefix, int maxLen) {
+        return namer.randomName(prefix, maxLen);
     }
```

Comment:
At this point it is more verbose for people to call this method that to simply call the code that this method calls. I would suggest removing this method entirely.